pkg/es/config/config.go

```diff
@@ -248,7 +250,8 @@ func (c *Configuration) IsEnabled() bool {
 
 // getConfigOptions wraps the configs to feed to the ElasticSearch client init
 func (c *Configuration) getConfigOptions() ([]elastic.ClientOptionFunc, error) {
-	options := []elastic.ClientOptionFunc{elastic.SetURL(c.Servers...), elastic.SetSniff(c.Sniffer)}
+	options := []elastic.ClientOptionFunc{elastic.SetURL(c.Servers...), elastic.SetSniff(c.Sniffer),
+		elastic.SetHealthcheck(!c.AllowTokenFromContext)}
```

Comment:
This deserves at least a comment: why is the healthcheck option the opposite of the "allow token from context" option? They don't really seem related to me... Perhaps you are disabling, because the health check endpoint requires auth, and the client doesn't have a way to obtain a token by itself?

Reply:
Yes, the reason is that if I don't have a valid token, the health check will fail, preventing query service to start. If I'm expecting a token from the other source, in this case from the header of the http request, I think it make sense to defer or disable the health check. 


Other thing I can do is trying to do the health check with the loaded token from a file (if any).

Reply:
Can you convert that into a succinct code comment?

Reply:
Done :)